libp2p/Host.cpp

```diff
@@ -149,12 +149,22 @@ void Host::stop()
 		return;
 
 	// wait for m_timer to reset (indicating network scheduler has stopped)
-	while (!!m_timer)
-		this_thread::sleep_for(chrono::milliseconds(50));
+	{
```

Comment:
Maybe we could call `m_timer.wait()` or `m_timer.cancel()` instead of all this piece? I'm not sure

Reply:
`m_timer` is a unique_ptr and it's reset in another thread.  I started using a condition variable.

Reply:
I think I have a commit containing a condition variable somewhere.  I have to recover that.